### PR TITLE
statement: initial support for common table expressions

### PR DESCRIPTION
**Pull request by esatterwhite**
All supported dialects have common table expressions at this point.
This includes a first pass at including support for WITH statement on
all DML statements

This lacks support for recursive common table expressions currently

### CODE REVIEW

File: src/statement.js [hunk 1]
```diff
@@ -234,6 +258,19 @@ class Statement {
     return this._buildClause('ORDER BY', result.join(', '));
   }
 
+  _buildCTE() {
```
Comment (esatterwhite) [inline src/statement.js]: if there is a better / more flexible way to do this, let me knw

Comment (jails) [reply src/statement.js]: That's perfect, however would it be ok to add an extra space like so:
```js
return this._buildClause('WITH', queries.join(', ')).trim() + ' ';
```
So there will be a space between the CTE end parenthesis and `SELECT/INSERT/UPDATE/DELETE` statements ?

**Comment by jails** [comment]
That's huge work, nice job :+1:

File: src/statement.js (hunk 1 above)
Comment (esatterwhite) [reply src/statement.js]: will do

Comment (esatterwhite) [reply src/statement.js]: fixed

**Pull request merged**